File: odroid/Share/src/bert/share/controller/Controller.java
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 *                 MIT License.
 */
package bert.share.controller;

import bert.share.message.MessageBottle;

/**
 *  A common interface for controllers owned by application instances.
 */
public interface Controller  {
	public void receiveRequest(MessageBottle request);
	public void receiveResponse(MessageBottle response);
	public void start();
	public void stop();
}

File: Bert/common/src/main/kotlin/chuckcoughlin/bert/share/controller/Controller.kt
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.share.controller

import bert.share.message.MessageBottle

/**
 * A common interface for controllers owned by application instances.
 */
interface Controller {
    fun receiveRequest(request: MessageBottle?)
    fun receiveResponse(response: MessageBottle?)
    fun start()
    fun stop()
}